Corgibytes.Freshli.Cli.Test/CommandOptions/AnalyzeCommandOptionsTest.cs

```diff
@@ -0,0 +1,104 @@
+using Corgibytes.Freshli.Cli.Test.Common;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Corgibytes.Freshli.Cli.Test.CommandOptions;
+
+[IntegrationTest]
+public class AnalyzeCommandOptionsTest : FreshliTest
+{
+    private const string ExpectedGitPath = "git";
+    private const string ExpectedBranch = "";
+    private const int ExpectedWorkerCount = 0;
+    private const string ExpectedHistoryInterval = "1m";
+    private const bool ExpectedCommitHistory = false;
+
+    public AnalyzeCommandOptionsTest(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    public static IEnumerable<object[]> AnalyzeOptionsArgs =>
+        new List<object[]>
+        {
+            // If passing no arguments, the default git path should be 'git'
+            new object[]
+            {
+                new[] { "analyze" }, "git", ExpectedBranch, ExpectedCommitHistory, ExpectedHistoryInterval,
+                ExpectedWorkerCount
+            },
+            // Specific git path expected
+            new object[]
+            {
+                new[] { "analyze", "--git-path", "/usr/bin/local/git" }, "/usr/bin/local/git", ExpectedBranch,
+                ExpectedCommitHistory, ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Specific branch expected
+            new object[]
+            {
+                new[] { "analyze", "--branch", "feature-fix-final.2.0" }, ExpectedGitPath, "feature-fix-final.2.0",
+                ExpectedCommitHistory, ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Entire commit history expected
+            new object[]
+            {
+                new[] { "analyze", "--commit-history" }, ExpectedGitPath, ExpectedBranch, true,
+                ExpectedHistoryInterval, ExpectedWorkerCount
+            },
+            // Three yearly history interval expected
+            new object[]
+            {
+                new[] { "analyze", "--history-interval", "3y" }, ExpectedGitPath, ExpectedBranch,
+                ExpectedCommitHistory, "3y", ExpectedWorkerCount
+            },
+            // 24 workers expected
+            new object[]
+            {
+                new[] { "analyze", "--workers", "24" }, ExpectedGitPath, ExpectedBranch, ExpectedCommitHistory,
+                ExpectedHistoryInterval, 24
+            }
+        };
+
+    [Theory]
+    [MemberData(nameof(AnalyzeOptionsArgs))]
+    public void Send_Args_ReturnsAnalyzeOptions(
+        string[] args, string expectedGitPath, string expectedBranch, bool expectedCommitHistory,
+        string expectedHistoryInterval, int expectedWorkers
+    )
+    {
+        var cmBuilder = Program.CreateCommandLineBuilder();
+        var parser = new Parser(cmBuilder.Command);
+
+        var result = parser.Parse(args);
+
+        var gitPath = result.GetOptionValueByName<string>("git-path");
+        gitPath?.Should().NotBeEmpty().And.Be(expectedGitPath);
```

Comment:
If `gitPath` is null then there's no assertion made against it's value. That doesn't seem correct to be. Seems like it doesn't properly test it's default value.

Reply:
That's definitely not correct. I didn't realize that. Thanks! 
The only value that is nullable is branch. https://github.com/corgibytes/freshli-cli/pull/212/commits/cb9a824809fa09908059b1779e63e790c6262be0 this commit reflects that.